lib/ex_aws/cloudfront/utils.ex

```diff
@@ -0,0 +1,39 @@
+defmodule ExAws.CloudFront.Utils do
+  alias ExAws.CloudFront.Policy
+
+  @doc """
+  Create a Signed URL Using a Policy and Query Builder.
+  """
+  def get_signed_url(policy, query_builder) do
+    policy.url
+    |> URI.parse
+    |> Map.update!(:query, fn query ->
+      query
+      |> to_string
+      |> URI.query_decoder
+      |> Stream.concat(policy |> Policy.to_statement |> Poison.encode! |> query_builder.())
+      |> URI.encode_query
+    end)
+    |> to_string
+  end
+
+  def get_signed_cookies(policy, cookies_builder) do
+    policy |> Policy.to_statement |> Poison.encode! |> cookies_builder.()
```

Comment:
You can't hardcode the JSON encoder here. ExAws has a very flexible policy on JSON codecs. You'll need to use the user supplied config. See the S3 presigned URLs for additional examples.